core-tests/shared/src/test/scala/zio/CauseSpec.scala

```diff
@@ -169,6 +171,77 @@ object CauseSpec extends ZIOBaseSpec {
         assert(stripped)(isNone)
       }
     ),
+    suite("toString")(
+      test("not fail with StackOverflowError") {
+        @tailrec
+        def genCause[E](current: Cause[E], depth: Int): Cause[E] =
+          if (depth <= 0) {
+            current
+          } else {
+            genCause(Both(current, Empty), depth - 1)
+          }
```

Comment:
I'm working on something separate that would make this test obsolete if left like this, so let's change it to this:

```suggestion
        def genCause(current: Cause[String], depth: Int): Cause[String] =
          if (depth <= 0) {
            current
          } else {
            genCause(Both(current, Cause.fail(s"Error$depth")), depth - 1)
          }
```